packages/sourcecred/src/util/batch.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+
+/**
+Returns an array of arrays that contains all of the items in the original
+array parameter, but batched into arrays no larger than the batchSize.
+Example:
+  batch([1,2,3,4,5], 2) = [[1,2], [3,4], [5]]
+*/
+export function batchArray<T>(
+  array: $ReadOnlyArray<T>,
+  batchSize: number
+): $ReadOnlyArray<$ReadOnlyArray<T>> {
+  const result = [];
+  let backlog = array;
+  while (backlog.length) {
+    result.push(backlog.slice(0, batchSize));
```

Comment:
pedantically speaking: what if the `batchSize` is `0` or with minus value? 
I think, the `while` loop will run forever, if the `batchSize` is smaller than `1`, for sure if `0`.

Reply:
Good point. I'll add a conditional throw.